Ginger/Ginger/Variables/VariableEditPages/VariableNumber.xaml.cs

```diff
@@ -0,0 +1,119 @@
+    {
+        VariableNumber variableNumber;
+        public VariableNumberPage(VariableNumber varNumber)
+        {
+            variableNumber = varNumber;
+            InitializeComponent();
+            GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(txtNumberValue, TextBox.TextProperty, varNumber, nameof(VariableNumber.InitialNumberValue));
+            txtNumberValue.AddValidationRule(new NumberValidationRule());
+            GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(rdoInputDecimal, RadioButton.IsCheckedProperty, varNumber, nameof(VariableNumber.IsDecimalValue));
+            GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(txtMinValue, TextBox.TextProperty, varNumber, nameof(VariableNumber.MinValue));
+            GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(txtMaxValue, TextBox.TextProperty, varNumber, nameof(VariableNumber.MaxValue));
+            GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(txtPrecisionValue, TextBox.TextProperty, varNumber, nameof(VariableNumber.PrecisionValue));
+            SetNummericType();
+
+            if (variableNumber.InitialNumberValue != null)
+            {
+                txtNumberValue.Text = variableNumber.InitialNumberValue;
+            }
+        }
+
+        private void SetNummericType()
+        {
+            if (!variableNumber.IsDecimalValue)
+            {
+                rdoInputInt.IsChecked = true;
+                pnlPrecision.Visibility = System.Windows.Visibility.Collapsed;
+                if (variableNumber.MinValue == null)
+                {
+                    variableNumber.MinValue = Int32.MinValue;
+                }
+
+                if (variableNumber.MaxValue == null)
+                {
+                    variableNumber.MaxValue = Int32.MaxValue;
+                }
+
+            }
+            else
+            {
+                rdoInputDecimal.IsChecked = true;
+                pnlPrecision.Visibility = System.Windows.Visibility.Visible;
+            }
+
+        }
+
+        private void rdoInputDecimal_Click(object sender, System.Windows.RoutedEventArgs e)
+        {
+            pnlPrecision.Visibility = System.Windows.Visibility.Visible;
+            variableNumber.MinValue = float.MinValue;
+            variableNumber.MaxValue = float.MaxValue;
+            if (variableNumber.PrecisionValue == null)
+            {
+                variableNumber.PrecisionValue = 2; //default
+            }
+        }
+
+        private void rdoInputInt_Click(object sender, System.Windows.RoutedEventArgs e)
+        {
+            pnlPrecision.Visibility = System.Windows.Visibility.Collapsed;
+            variableNumber.MinValue = Int32.MinValue;
+            variableNumber.MaxValue = Int32.MaxValue;
+        }
+
+        private void txtNumberValue_LostFocus(object sender, System.Windows.RoutedEventArgs e)
+        {
+            //txtNumberValue.BindControl(variableNumber,nameof(VariableNumber.InitialNumberValue));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5355677)